aclk/aclk_rx_msgs.c

```diff
@@ -15,18 +15,52 @@ static inline int aclk_extract_v2_data(char *payload, char **data)
     return 0;
 }
 
-static inline int aclk_v2_payload_get_query(const char *payload, struct aclk_request *req)
+#define ACLK_GET_REQ "GET "
+#define ACLK_CHILD_REQ "/host/"
+#define ACLK_CLOUD_REQ_V2_PREFIX "/api/v1/"
+#define STRNCMP_CONST(str, const) strncmp(str, const, strlen(const))
+static inline int aclk_v2_payload_get_query(struct aclk_cloud_req_v2 *cloud_req, struct aclk_request *req)
 {
-    const char *start, *end;
+    const char *start, *end, *ptr;
+    char uuid_str[UUID_STR_LEN];
+    uuid_t uuid;
 
-    if(strncmp(payload, ACLK_CLOUD_REQ_V2_PREFIX, strlen(ACLK_CLOUD_REQ_V2_PREFIX))) {
-        errno = 0;
+    errno = 0;
+
+    if(STRNCMP_CONST(cloud_req->data, ACLK_GET_REQ)) {
+        error("Only accepting GET HTTP requests from CLOUD");
+        return 1;
+    }
+    start = ptr = cloud_req->data + strlen(ACLK_GET_REQ);
+
```

Comment:
Should we avoid / optimize these strlen functions on the defined strings?

Reply:
the compiler will turn that into constant AFAIK

Reply:
yep I remember that correctly `strlen` on literal/constant string is resolved at compile time and turned into constant.

Reply:
yep I checked you can make small program like this:
```
#include <stdio.h>
#include <string.h>

#define WAKA "WakaWaka"

int main() {
        printf("Len %d %s\n", strlen(WAKA), WAKA);
}
```
and compile with gcc -S test.c
From generated assembly output you will see `strlen(WAKA)` is replaced by constant 8